src/index.tsx

```diff
@@ -89,7 +91,7 @@ const calculateTotalItems = (items = []) =>
 
 const calculateUniqueItems = (items = []) => items.length;
 
-export function CartProvider({
+export const CartProvider: React.FC<any> = ({
```

Comment:
It's also worth us typing the props available here too.

Reply:
Yes, VERY much this, because this will be exposed directly to the user of the lib.
We'll also need to make a nice type for the `storage` variable, that will be able to know a lot about what it is supposed to handback, to the CartProvider, and what the function is expected to take.

Reply:
I'll make a PR to resolve this.

Reply:
If we had a `CartProviderProps` type then I imagine the `storage` would just be `key: string, initialState: object`

Reply:
Yes, but the storage is able to take 
storage = somefunction(key: string, initialState: object) => returns [savedCart, saveCart], right?